build-common.xml

```diff
@@ -288,7 +366,8 @@
       </antcontrib:if>
     </target>
 
-    <target name="test" depends="test-compile" description="Run unit tests">
+    <!-- Main test target: single JUnit run wrapped in Jacoco (no double JUnit here) -->
+    <target name="test" depends="test-compile,define-jacoco-tasks" description="Run unit tests and generate JaCoCo coverage">
```

Comment:
lets separate out the dependency on coverage to skip if required to only execute test